### DISCUSSION TITLE
Add Packing options to `PredictionRigidbody`

### DISCUSSION THREAD

**Discussion by AustinEast** [FirstGearGames/FishNet#823] [Feature Request]
Im running into issues with accuracy when reconciling my Predicted Rigidbodies, and I've narrowed it down to the `RigidbodyState` reconciling with a packed Quaternion - after manually updating the `RigidBodyState` writer/reader to use unpacked quaternions my issues are resolved.

So my request would either be to add an option to set the Packing on the PredictionRigidbody/RigidbodyState, or to expose the internal `PredictionRigidbody.SetReconcileData()` method so that we can easily write our own writers/readers for the type (the mentioned internal method is the main blocker for this DIY approach).

**Comment by AustinEast** [comment]
Update - I looked into the rotation quaternion issue I've been seeing, and it seems like it only happens on the client when i have `Create Local Reconciles` active and I'm not reconciling every frame from the server - there may be inconsistencies happening between the packed rotation quaternions from the server reconcile states and what i assume are untouched rotation quaternions from the local reconcile states. 

FWIW I haven't proven this theory yet, just food for thought.

**Comment by FirstGearGames** [comment]
I recall we talked about this. I believe a packing option makes sense.

There's some new Quaternion packing stuff I worked on that will be perfect for this; you essentially specify precision and it packs based on that. What is normally 8 bytes is ~6.2, and 12 bytes ~ 6.7.  I'll try to get this in the 4.6.0 release.

**Comment by FirstGearGames** [comment]
The new packing is live but not in PredictedRigidbody. I'd like to have you test if you can DM me please.
